src/Education/ErrorList/SonarErrorListEventProcessor.cs

```diff
@@ -56,9 +56,10 @@ public override void PreprocessNavigateToHelp(
 
                 if (errorListHelper.TryGetRuleId(entry, out var ruleId))
                 {
+                    errorListHelper.TryGetFilterableIssue(entry, out var filterableIssue);
                     logger.LogVerbose(Resources.ErrorList_Processor_SonarRuleDetected, ruleId);
 
-                    educationService.ShowRuleHelp(ruleId, /* todo */ null);
+                    educationService.ShowRuleHelp(ruleId, null, filterableIssue?.IssueId);
```

Comment:
Removed todo without fixing the underlying issue of rule context not being added. If now all issues get the required rule context from `getEffectiveIssueDetails`, with the exception of CFamily, which anyway will soon be migrated,  and this todo is no longer relevant, then this parameter can be removed along with the todo.

Reply:
I have no clue what is the underlying cause of the todo. And we have all over the place todos that do not explain at all what should be done and in which task. 
I think this should be taken in a different PR.

Reply:
Added back the todo and linked it and similar todos to corresponding task (https://sonarsource.atlassian.net/browse/SLVS-1630)